src/main/java/core/basesyntax/utils/process/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.utils.process.impl;
+
+import core.basesyntax.storage.impl.StorageImpl;
+import core.basesyntax.utils.process.ShopService;
+import core.basesyntax.utils.transaction.FruitTransaction;
+import core.basesyntax.utils.transaction.handlers.OperationHandlerFactory;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationHandlerFactory factory;
+    private final StorageImpl storage;
```

Comment:
Avoid using concrete classes for fields like `StorageImpl` here. Instead, use the interface type to follow the Dependency Inversion Principle. This would make your code more flexible and easier to test.